x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/PivotConfig.java

```diff
@@ -29,6 +30,7 @@
     private static final String NAME = "data_frame_transform_pivot";
     private final GroupConfig groups;
     private final AggregationConfig aggregationConfig;
+    private final Integer size;
```

Comment:
On having `size` in pivot instead of the transform object itself:

I think the decision on this depends on whether we imagine that a transform will always use a composite aggregation. If yes, then `size` should be on the top level.